addons/web/static/src/core/datetime/datetime_picker.xml

```diff
@@ -74,8 +77,10 @@
                     </t>
                 </t>
             </div>
-
-            <div t-attf-class="o_datetime_buttons position-md-absolute end-0">
+            <div
+                class="o_datetime_buttons end-0"
+                t-att-class="{ 'position-md-absolute': props.type === 'datetime'  }"
+            >
```

Comment:
This is the only useful diff in this file.

The other changes are only linting/refactoring and I am not comfortable seeing those kind of changes in a stable branch, even if they are minor:
- first it kind of "obfuscate" the real fix
- then it could potentially break a user customization (thin chances but still...)